adapter/internal/operator/controllers/dp/api_controller.go

```diff
@@ -218,10 +218,13 @@ func (apiReconciler *APIReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 // - HTTPRoutes
 func (apiReconciler *APIReconciler) resolveAPIRefs(ctx context.Context, namespace string,
 	prodHTTPRouteRef string, sandHTTPRouteRef string) (*synchronizer.HTTPRouteState, *synchronizer.HTTPRouteState, error) {
-	var prodHTTPRoute *synchronizer.HTTPRouteState
-	var sandHTTPRoute *synchronizer.HTTPRouteState
+	prodHTTPRoute := &synchronizer.HTTPRouteState{
```

Comment:
This `non nil initialization` added to update the internal api cache. In scenario when prod or sand `HTTPRoute` is added initial deployment and then commented out in the second deployment. Otherwise commenting out prod or sand ref does not undeploy the prod or sand.